plugins/inputs/ssl/check_ssl.go

```diff
@@ -0,0 +1,187 @@
+package ssl
+
+import (
+	"crypto/sha1"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var signatureAlgorithm = [...]string{
+	"UnknownSignatureAlgorithm",
+	"MD2WithRSA",
+	"MD5WithRSA",
+	"SHA1WithRSA",
+	"SHA256WithRSA",
+	"SHA384WithRSA",
+	"SHA512WithRSA",
+	"DSAWithSHA1",
+	"DSAWithSHA256",
+	"ECDSAWithSHA1",
+	"ECDSAWithSHA256",
+	"ECDSAWithSHA384",
+	"ECDSAWithSHA512",
+}
+
+var publicKeyAlgorithm = [...]string{
+	"UnknownPublicKeyAlgorithm",
+	"RSA",
+	"DAS",
+	"ECDSA",
+}
+
+func SHA1Hash(data []byte) string {
+	h := sha1.New()
+	h.Write(data)
+	return fmt.Sprintf("%X", h.Sum(nil))
+}
+
+// SSLCerts struct
+type SSLCerts struct {
+	SHA1                string
+	SubjectKeyId        string
+	Version             int
+	SignatureAlgorithm  string
+	PublicKeyAlgorithm  string
+	Subject             string
+	DNSNames            []string
+	NotBefore, NotAfter string
+	ExpiresIn           int64
+	Issuer              string
+	AuthorityKeyId      string
+}
+
+type CheckExpire struct {
+	// Server to check
+	Servers []string
+
+	// SSL server port number (Default 443)
+	Port string
+
+	// Timeout in seconds. 0 means no timeout
+	Timeout int
```

Comment:
I might do a string instead, and then use `time.ParseDuration`. Aside from making it easier to read the config (looking at a value `1s`, you know what the unit is), it also allows you to specify sub-second timeout. The alternative might be to use milliseconds instead.